rep-0150.rst

```diff
@@ -0,0 +1,210 @@
+ * desktop
+ * ros_base
+
+Metapackages
+------------
+
+ROS Core
+''''''''
+
+The `ros_core` metapackage composes the core communication protocols.
+It may not contain any GUI dependencies.
+
+::
+
+ - ros_core:
+      extends: [ros, ros_comm]
+      packages: [catkin, cmake_modules, common_msgs, console_bridge,
+                 gencpp, geneus, gennodejs, genlisp, genmsg, genpy,
+                 message_generation, message_runtime, rosbag_migration_rule,
+                 rosconsole_bridge, roscpp_core, rosgraph_msgs, roslisp,
+                 rospack, std_msgs, std_srvs]
+
+ROS Base
+''''''''
+
+The `ros_base` metapackage composes the `ros_core` metapackage with
+commonly used libraries.
+It may not contain any GUI dependencies.
+
+::
+
+  - ros_base:
+      extends: [ros_core]
+      packages: [actionlib, angles, bond_core, class_loader,
+                 dynamic_reconfigure, nodelet_core,
+                 pluginlib]
+
+Robot metapackage
+'''''''''''''''''
+
+The `robot` metapackage is defined to be core, stable, ROS libraries
+for any robot hardware.
+It is the "general robotics" libraries of ROS.
+It may not contain any GUI dependencies.
+In ROS Melodic `robot` is contracted to remove `robot_model` and instead
+explicitly list the necessary packages.
+
+::
+
+  - robot:
+      extends: [ros_base]
+      packages: [collada_parser, control_msgs, diagnostics, executive_smach,
+                 filters, geometry, joint_state_publisher, kdl_parser,
+                 kdl_parser_py, robot_state_publisher, urdf,
+                 urdf_parser_plugin, xacro]
+
+Capability variants
+'''''''''''''''''''
+
+The capability variants organize commonly used libraries that are
+specific to a class of use case.
+We also define a `simulators` variant that provides an organizational
+role for higher-level variants.
+We discourage GUI dependencies in these stacks, if possible.
+
+::
+
+  - perception:
+      extends: [ros_base]
+      packages: [image_common, image_pipeline,
+                 image_transport_plugins, laser_pipeline,
+                 perception_pcl, vision_opencv]
+
+  - simulators:
+      extends: [robot]
+      packages: [gazebo_ros_pkgs, rqt_common_plugins, rqt_robot_plugins,
+                 stage_ros]
+
+  - viz:
+      extends: [robot]
+      packages: [rqt_common_plugins, rqt_robot_plugins, rviz]
+
+Desktop variants
+''''''''''''''''
+
+The `desktop` variants are main entry points for users.
+The `desktop_full` is a "batteries included" experience for users and
+attempts to collect stable, well-documented libraries.
+These libraries may be specific to certain classes of robots, such as
+mobile robots, though they are not specific to a particular robot.
+The `desktop` variant is more minimal and only provides the stacks in
+the `robot` variant, plus visualization and debugging tools.
+Both of these variants contain tutorials for the stacks they provide.
+`urdf_tutorials` are only in `desktop_full` because they depend on the simulator.
```

Comment:
`urdf_tutorials` << this is still not an existing package or variant. maybe use `urdf_sim_tutorial` here? or a non-quoted name like "the urdf tutorials are in `desktop_full` ..."

Reply:
Yeah, good point.  Will change to the latter without quotes.